### ISSUE TITLE
Some more false negative for testing-library/prefer-screen-queries

### ISSUE THREAD

**Issue by julienw** [testing-library/eslint-plugin-testing-library#391]
Hey @Belco90
I just upgraded to the new version that fixed #367, and I am reporting that it indeed reported new cases...but not all of them!

For example these cases aren't reported still: https://github.com/firefox-devtools/profiler/blob/1c5dfed055f864413b3a62e43ca2d27b074d6108/src/test/components/CallNodeContextMenu.test.js#L107-L112

I'm pretty sure this is because of the spread operator. And indeed I reproduced by slightly adjusting one of the smaller testcases:
```js
import * as React from 'react';
import { render } from 'firefox-profiler/test/fixtures/testing-library';

function setup() {
  const result = render(<div />);
  return { ...result };
}

it('detected', async () => {
  const { getByText } = await setup();

  expect(getByText('foo')).toBeInTheDocument(); // detected
});

it('undetected', () => {
  const { getByText } = setup();

  expect(getByText('foo')).toBeInTheDocument(); // undetected
});
```

Hope this helps :-)

**Comment by Belco90** [comment]
Hey @julienw! Thanks for reporting more feedback about `prefer-screen-queries` rule. It definitely seems the rule is not ready for the spread operator.

**Comment by julienw** [comment]
If I understand properly the current implementation for the rule, it tries to follow where variables were declared, so that when there's a call with eg `getByText` the rule can decide if the variable is legit. My understanding is that the rule minimizes the false positives. I would like to suggest that it could minimize false negatives instead: the reason is that it's easy to silence false positives, but false negatives are forever lost.

Therefore, why can't we report all `getByText` calls that are "standalone" to start with (that is they're not method calls) ? This would also follow the spirit of not fixing #143.
Possible false positives is if a project has `getByText` and friends functions that aren't from testing-library.
